vcd/resource_vcd_security_tag.go

```diff
@@ -0,0 +1,113 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdOpenApiSecurityTag() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdOpenApiSecurityTagCreate,
+		ReadContext:   resourceVcdOpenApiSecurityTagRead,
+		UpdateContext: resourceVcdOpenApiSecurityTagCreate,
+		DeleteContext: resourceVcdOpenApiSecurityTagDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdOpenApiSecurityTagImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Security tag name to be created",
+			},
+			"vm_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "List of VM IDs that the security tags is going to be tied to",
+				MinItems:    1, // If vm_ids has nothing, the tag will be removed. We enforce to have at least 1 to avoid that behavior
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceVcdOpenApiSecurityTagCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	securityTagName := d.Get("name").(string)
+
+	securityTag := &types.SecurityTag{
+		Tag:      securityTagName,
+		Entities: convertSchemaSetToSliceOfStrings(d.Get("vm_ids").(*schema.Set)),
+	}
+	_, err := vcdClient.UpdateSecurityTag(securityTag)
+	if err != nil {
+		return diag.Errorf("error when setting up security tags - %s", err)
+	}
+
+	d.SetId(securityTagName)
+
+	return resourceVcdOpenApiSecurityTagRead(ctx, d, meta)
+}
+
+func resourceVcdOpenApiSecurityTagRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	securityTagName := d.Id()
+	taggedEntities, err := vcdClient.GetSecurityTaggedEntities(fmt.Sprintf("tag==%s", securityTagName))
+	if err != nil {
```

Comment:
I must remind the case of handling the error if the object is deleted outside of Terraform 

https://github.com/vmware/terraform-provider-vcd/pull/800#discussion_r825335060

Probably the most fluent way to achieve this would be to have a dedicated function `vcdClient.GetSecurityTaggedEntityByName` which returns `govcd.ErrorEntityNotFound` if that exact item is not found.

Reply:
My fault. Added a new function on govcd called `Org.GetAllSecurityTaggedEntityByName` that wraps `Org.GetAllSecurityTaggedEntities` and returns `ErrorEntityNotFound` if nothing was found.
Also sets the Id to empty if that error is received.